common4j/src/main/com/microsoft/identity/common/java/authorities/Authority.java

```diff
@@ -170,12 +178,21 @@ public static Authority getAuthorityFromAuthorityUrl(String authorityUrl) {
                     authority = new AzureActiveDirectoryB2CAuthority(authorityUrl);
                     break;
                 default:
-                    Logger.verbose(
-                            TAG + methodName,
-                            "Authority type default: AAD"
-                    );
-                    authority = createAadAuthority(authorityCommonUriBuilder, pathSegments);
-                    break;
+                    if (authorityUrl.contains(CIAMAuthority.CIAM_LOGIN_URL_SEGMENT)) {
+                        Logger.verbose(
+                                TAG + methodName,
+                                "Authority type is CIAM"
+                        );
+                        authority = new CIAMAuthority(authorityUrl);
+                        break;
+                    } else {
```

Comment:
Consider moving this out of switch case

Reply:
Ended up getting rid of the switch statement